### PR TITLE
Migrates from tut to mdoc

### PR DESCRIPTION
**Pull request by juanpedromoreno**
## What this does?

This PR migrates from tut to mdoc.

## Checklist

- [x] Reviewed the diff to look for typos, println and format errors.
- [x] Updated the docs accordingly.

### CODE REVIEW

File: project/ProjectPlugin.scala [hunk 1]
```diff
@@ -316,9 +317,13 @@ object ProjectPlugin extends AutoPlugin {
       micrositeHighlightLanguages += "protobuf"
     )
 
+    lazy val customScalacOptions: Seq[String] = scalacAdvancedOptions.filterNot(Set("-Yliteral-types", "-Xlint").contains)
+    lazy val docsExclusions: Seq[String] => Seq[String] = (current: Seq[String]) => current.filterNot(Set("-Xfatal-warnings", "-Ywarn-unused-import", "-Xlint").contains)
+
     lazy val docsSettings: Seq[Def.Setting[_]] = Seq(
       libraryDependencies += %%("scalatest", V.scalatest),
-      scalacOptions in Tut ~= (_ filterNot Set("-Ywarn-unused-import", "-Xlint").contains)
+      mdocExtraArguments ++= Seq("--no-link-hygiene"),
```
Comment (BenFradet) [inline project/ProjectPlugin.scala]: out of curiosity, why was this needed?

Comment (juanpedromoreno) [reply project/ProjectPlugin.scala]: Removed in 6e2f855

**Comment by juanpedromoreno** [comment]
After rebasing against the master branch, it seems like scalafmt 2.4.1 introduced a lot of formatting changes in this PR :/

**Comment by cb372** [comment]
I got this on the "Patterns" page when I ran `docs/makeMicrosite` locally.

Screenshot 2020-02-17 at 13 12 41

**Comment by juanpedromoreno** [comment]
@cb372 mmm weird, it works for me locally. Did you clean out the target? It's also working in Travis 🤔

**Comment by cb372** [comment]
@juanpedromoreno Yes, this is after cleaning. Just to clarify, this is not in the log output, it's actually displayed on the site. 

```
sbt clean docs/makeMicrosite
cd docs/target/site
jekyll serve -b /mu-scala
open 'http://127.0.0.1:4000/mu-scala/patterns#service-testing'
(scroll down a bit)
```

**Comment by juanpedromoreno** [comment]
Thanks @cb372! I think that I've fixed the issue in the renderer HTML by removing the useless code. Could you please take a second look?

**Comment by cb372** [comment]
Thanks @juanpedromoreno , it works now! I just added a commit to tweak the scalatest output.
